### PR TITLE
[geographiclib] Geographiclib/2.4

### PR DESCRIPTION
**Pull request by cffk**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/geographiclib/portfile.cmake [hunk 1]
```diff
@@ -2,9 +2,8 @@ vcpkg_from_sourceforge(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO geographiclib
     REF distrib-C++
-    FILENAME "GeographicLib-2.3.tar.gz"
-    SHA512 1a1bd0fc2dc3e1372cf22618af3a4340bbc6497f94c64226c97654dfff92a4bf3acf47d91592741fe0c643d401d9721f680bdb4974b8ee258fb09d525fbaec67
-    PATCHES remove-broken-and-unnecessary-cross-compile-check.patch
+    FILENAME "GeographicLib-2.4.tar.gz"
```
Comment (WangWeiLin-MV) [inline ports/geographiclib/portfile.cmake:5]:
```suggestion
    FILENAME "GeographicLib-${VERSION}.tar.gz"
```

File: ports/geographiclib/remove-broken-and-unnecessary-cross-compile-check.patch
Comment (WangWeiLin-MV) [inline ports/geographiclib/remove-broken-and-unnecessary-cross-compile-check.patch:1]: Info: The patch has been merged in https://github.com/geographiclib/geographiclib/commit/1b0be832df51665ebe943f6d4d72eabc0de1bb92.

File: ports/geographiclib/portfile.cmake (hunk 1 above)
Comment (cffk) [reply ports/geographiclib/portfile.cmake]: Done

File: ports/geographiclib/vcpkg.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "geographiclib",
-  "version": "2.3",
+  "version": "2.4",
   "port-version": 1,
```
Comment (JavierMatosD) [inline ports/geographiclib/vcpkg.json:4]: port-version should be reset on version update. 
```suggestion
```

File: versions/g-/geographiclib.json [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "eb8afcc9843755d37ecfa35b6e211b67c439b49b",
+      "version": "2.4",
+      "port-version": 1
+    },
+    {
+      "git-tree": "f94e42e48c15b3fa4dbc4e8412c74dd53608cf5d",
+      "version": "2.4",
+      "port-version": 0
+    },
```
Comment (JavierMatosD) [inline versions/g-/geographiclib.json:12]: There should only be one new entry.
```suggestion
    {
      "git-tree": "f94e42e48c15b3fa4dbc4e8412c74dd53608cf5d",
      "version": "2.4",
      "port-version": 0
    },
```

After committing all the requested changes, run `vcpkg x-add-version geographiclib.json --overwrite-version`

**Review by JavierMatosD — DISMISSED** [review]

**Comment by JavierMatosD** [comment]
Mark "Ready for Review" once you've applied the changes.

File: versions/g-/geographiclib.json (hunk 3 above)
Comment (cffk) [reply versions/g-/geographiclib.json]: OK, done

File: ports/geographiclib/vcpkg.json (hunk 2 above)
Comment (cffk) [reply ports/geographiclib/vcpkg.json]: Done.

**Pull request merged**